content/transfer-fee.md

```diff
@@ -0,0 +1,959 @@
+ - Assigns to it's owning program
+
+As with all Token Extension Program's mints, we need to calculate the space and lamports needed for the mint. We can get these by calling `getMintLen` and `getMinimumBalanceForRentExemption`
+
+```ts
+const extensions = [ExtensionType.TransferFeeConfig]
+const mintLength = getMintLen(extensions)
+
+const mintLamports =
+	await connection.getMinimumBalanceForRentExemption(mintLength)
+
+const createAccountInstruction = SystemProgram.createAccount({
+	fromPubkey: payer.publicKey,
+	newAccountPubkey: mintKeypair.publicKey,
+	space: mintLength,
+	lamports: mintLamports,
+	programId: TOKEN_2022_PROGRAM_ID,
+})
+```
+
+The second instruction `createInitializeTransferFeeConfigInstruction` initializes the transfer fee extension.
+
+It takes the following parameters:
+- `mint`: Token mint account
+- `transferFeeConfigAuthority`: Optional authority that can update the fees
+- `withdrawWithheldAuthority`: Optional authority that can withdraw fees
+- `transferFeeBasisPoints`: Amount of transfer collected as fees, expressed as basis points of the transfer amount
+- `maximumFee`: Maximum fee assessed on transfers
+- `programId`: SPL Token program account
+
+```ts
+const initializeTransferFeeConfigInstruction = createInitializeTransferFeeConfigInstruction(
+	mintKeypair.publicKey,
+	payer.publicKey,
+	payer.publicKey,
+	feeBasisPoints,
+	maxFee,
+	TOKEN_2022_PROGRAM_ID
+)
+```
+
+The third instruction `createInitializeMintInstruction` initializes the mint.
+
+```ts
+const initializeMintInstruction = createInitializeMintInstruction(
+	mintKeypair.publicKey,
+	decimals,
+	payer.publicKey,
+	null,
+	TOKEN_2022_PROGRAM_ID
+)
+```
+
+Lastly, you need to add all of these instructions to a transaction and send it off the the blockchain.
+
+```ts
+const mintTransaction = new Transaction().add(
+	createAccountInstruction,
+	initializeTransferFeeConfigInstruction,
+	initializeMintInstruction
+);
+
+const signature = await sendAndConfirmTransaction(
+	connection,
+	mintTransaction,
+	[payer, mintKeypair],
+	{commitment: 'finalized'}
+)
+```
+
+## Transferring mint with transfer fees
+
+There are a couple of notes when transferring tokens with the `transfer fee` extension.
+
+First, the recipient is the one who "pays" for the fee. If I send 100 tokens with basis points of 50 (5%), the recipient will receive 95 tokens (five withheld)
+
+Second, the fee is calculated not by the tokens sent, but the smallest unit of said token. In Solana programming, we always specify amounts to be transferred, minted or burned in their smallest unit. To send one SOL to someone, we actually send `1 * 10 ^ 9` lamports. Another way to look at it is if you wanted to send one US dollar, you're actually sending 100 pennies. Let's make this dollar a token with a 50 basis points (5%) transfer fee. Sending one dollar, would result in a five cent fee. Now let's say we have a max fee of 10 cents, this will always be the highest fee, even if we send $10,000.
```

Comment:
```suggestion
Second, the fee is calculated not by the tokens sent, but the smallest unit of said token. In Solana programming, we always specify amounts to be transferred, minted or burned in their minor unit. To send one SOL to someone, we actually send `1 * 10 ^ 9` lamports. Another way to look at it is if you wanted to send one US dollar, you're actually sending 100 pennies. Let's make this dollar a token with a 50 basis points (5%) transfer fee. Sending one dollar, would result in a five cent fee. Now let's say we have a max fee of 10 cents, this will always be the highest fee, even if we send $10,000.
```